src/lib/operators/table_scan/abstract_dereferenced_column_table_scan_impl.hpp

```diff
@@ -43,6 +43,40 @@ class AbstractDereferencedColumnTableScanImpl : public AbstractTableScanImpl {
 
   const std::shared_ptr<const Table> _in_table;
   const ColumnID _column_id;
+
+  template <typename IteratorType>
+  void _handle_search_results(IteratorType begin, IteratorType end, const ChunkID chunk_id, PosList& matches,
+                              const std::shared_ptr<const PosList>& position_filter) const {
+    if (begin == end) return;
+
+    // General note: If the predicate is NotEquals, there might be two matching ranges. scan_sorted_segment
+    // combines these two ranges into a single one via boost::join(range_1, range_2).
+    // See sorted_segment_search.hpp for further details.
```

Comment:
This method is only used by sorted segment searches, correct? Move to `sorted_segment_search.hpp` (and remove this line)?